samples/generate-geodatabase-replica-from-feature-service/src/main/java/com/esri/arcgismaps/sample/generategeodatabasereplicafromfeatureservice/components/GenerateGeodatabaseReplicaFromFeatureServiceViewModel.kt

```diff
@@ -0,0 +1,318 @@
+     */
+    fun resetMap() {
+        // clear any layers  and symbols already on the map
+        arcGISMap.operationalLayers.clear()
+        graphicsOverlay.graphics.clear()
+        // add the download area boundary
+        graphicsOverlay.graphics.add(downloadArea)
+        // add back the feature layer
+        arcGISMap.operationalLayers.add(featureLayer)
+        // close the current geodatabase, if a replica was already generated
+        geodatabase?.close()
+        // show the Generate button
+        generateButtonEnabled = true
+        resetButtonEnabled = false
+    }
+
+    /**
+     * Generate the geodatabase replica.
+     */
+    fun generateGeodatabaseReplica() {
+        // disable the Generate button
+        generateButtonEnabled = false
+
+        val offlineGeodatabasePath =
+            application.getExternalFilesDir(null)?.path + "/portland_trees_gdb.geodatabase"
+
+        // delete any offline geodatabase already in the cache
+        File(offlineGeodatabasePath).deleteRecursively()
+
+        // get the geometry of the download area
+        val geometry = downloadArea.geometry ?: return messageDialogVM.showMessageDialog(
+            title = "Could not get geometry of the downloadArea"
+        )
+
+        viewModelScope.launch(Dispatchers.Main) {
+            // create GenerateGeodatabaseParameters for the selected extent
+            val parameters =
+                geodatabaseSyncTask.createDefaultGenerateGeodatabaseParameters(geometry).getOrElse {
+                    messageDialogVM.showMessageDialog(
+                        title = "Error creating geodatabase parameters",
+                        description = it.message.toString()
+                    )
+                    return@launch
+                }.apply {
+                    // modify the parameters to only include the Trees (0) layer
+                    layerOptions.removeIf { layerOptions ->
+                        layerOptions.layerId != 0L
+                    }
+                }
+
+            // we don't need attachments
+            parameters.returnAttachments = false
+
+            // create a GenerateGeodatabaseJob
+            val job = geodatabaseSyncTask.createGenerateGeodatabaseJob(
+                parameters = parameters,
+                pathToGeodatabaseFile = offlineGeodatabasePath
+            )
+
+            // stash the job so the cancel function can use it
+            generateGeodatabaseJob = job
+
+            // run the job
+            runGenerateGeodatabaseJob(job)
+        }
+    }
+
+    /**
+     * Run the [job], showing the progress dialog and displaying the resultant data on the map.
+     */
+    private suspend fun runGenerateGeodatabaseJob(job: GenerateGeodatabaseJob) {
+        // create a flow-collection for the job's progress
+        viewModelScope.launch(Dispatchers.Main) {
+            job.progress.collect { progress ->
+                jobProgress = progress
+            }
+        }
```

Comment:
Could expose `job.progress` to the UI directly as well - although there isn't always a generateGeodatabaseJob, so it may not be so simple.